### DISCUSSION TITLE
Cannot invite a new member

### DISCUSSION THREAD

**Discussion by Massimiliano-solutiontech** [dani-garcia/vaultwarden#3490] [Q&A] [answered]
### Cannot invite a new member
I can't invite a new member into an organization when the user doesn't exists

### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.28.1-39364561
* Web-vault version: v2023.3.0b
* OS/Arch: linux/x86_64
* Running within Docker: false (Base: Not applicable)
* Environment settings overridden: false
* Uses a reverse proxy: true
* IP Header check: true (X-Real-IP)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Browser/Server Time Check: true
* Server/NTP Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: PostgreSQL
* Database version: PostgreSQL 14.7 (Ubuntu 14.7-0ubuntu0.22.04.1) on x86_64-pc-linux-gnu, compiled by gcc (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0, 64-bit
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** 

```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": false,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "_smtp_img_src": "cid:",
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_session_lifetime": 20,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "/opt/vaultwarden/data/attachments",
  "authenticator_disable_time_drift": true,
  "data_folder": "/opt/vaultwarden/data",
  "database_conn_init": "",
  "database_max_conns": 10,
  "database_timeout": 30,
  "database_url": "**********://**************************************************",
  "db_connection_retries": 15,
  "disable_2fa_remember": true,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://**************************",
  "domain_origin": "*****://**************************",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": false,
  "emergency_notification_reminder_schedule": "0 3 * * * *",
  "emergency_request_timeout_schedule": "0 7 * * * *",
  "enable_db_wal": true,
  "event_cleanup_schedule": "0 10 0 * * *",
  "events_days_retain": null,
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "/opt/vaultwarden/data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "invitation_expiration_hours": 120,
  "invitation_org_name": "Solution Tech",
  "invitations_allowed": false,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": null,
  "log_level": "Info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "",
  "org_events_enabled": false,
  "org_groups_enabled": false,
  "password_hints_allowed": false,
  "password_iterations": 600000,
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "/opt/vaultwarden/data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sendmail_command": null,
  "sends_allowed": true,
  "sends_folder": "/opt/vaultwarden/data/sends",
  "show_password_hint": false,
  "signups_allowed": false,
  "signups_domains_whitelist": "",
  "signups_verify": false,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_embed_images": true,
  "smtp_explicit_tls": null,
  "smtp_from": "**************************************",
  "smtp_from_name": "Vaultwarden",
  "smtp_host": "********************************",
  "smtp_password": "***",
  "smtp_port": 465,
  "smtp_security": "force_tls",
  "smtp_ssl": null,
  "smtp_timeout": 15,
  "smtp_username": "**************************************",
  "templates_folder": "/opt/vaultwarden/data/templates",
  "tmp_folder": "/opt/vaultwarden/data/tmp",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_sendmail": false,
  "use_syslog": false,
  "user_attachment_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "/opt/vaultwarden/web-vault",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": true,
  "websocket_port": 3012,
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```
</details>

* Install method: LXC (https://github.com/tteck/Proxmox)

* Clients used: web vault

* Reverse proxy and version: nginx

* MySQL/MariaDB or PostgreSQL version: PostgreSQL 14.7

### Steps to reproduce
1. go to an organization
2. members tab
3. click on invite member
4. insert an email
5. save

### Expected behaviour
Because the user doesn't exists I expect that an invitation email is sent to the above specified email

### Actual behaviour
I get a `400` HTTP error from `https://**************************/api/organizations/************/users/invite` with this payload:

```json
{
    "ErrorModel": {
        "Message": "User does not exist: *****@*********.***",
        "Object": "error"
    },
    "ExceptionMessage": null,
    "ExceptionStackTrace": null,
    "InnerExceptionMessage": null,
    "Message": "User does not exist: *****@*********.***",
    "Object": "error",
    "ValidationErrors": {
        "": [
            "User does not exist: *****@*********.***"
        ]
    },
    "error": "",
    "error_description": ""
}
```

**Comment by stefan0xC** [comment, marked as answer]
You don't have `invitations_allowed`.

**Comment by Massimiliano-solutiontech** [comment]
Thanks, it worked but I checked and in the `.env` I already had the `INVITATIONS_ALLOWED=TRUE` so I had to add the property to the `config.json` too. If the property is only in the `.env` shouldn't it use that?

**Reply by BlackDex** [reply]
The `config.json` overrules environment variables!

**Reply by Massimiliano-solutiontech** [reply]
Yes I know, but as I said before I had to set the key in the `config.json` too (in the `.env` was already present the key with the value to `true`). So I would like to know if it is a bug or I have something wrong with my configuration

**Reply by BlackDex** [reply]
You need to restart after editing the config via either environment variables or changing the `config.json`.

If that didn't worked, then either it was the wrong file, or the file was edited in a wrong way.

**Reply by Massimiliano-solutiontech** [reply]
I attached where the file is and the content. Maybe the location is wrong

![IMG_0504](https://user-images.githubusercontent.com/75322363/235779041-fbde88f3-66f0-4398-a431-295c61ab1ff6.jpeg)
![IMG_0505](https://user-images.githubusercontent.com/75322363/235779053-4e5f3fe3-e70e-4955-800d-31e52e89e956.jpeg)

**Reply by BlackDex** [reply]
That depends on how you start Vaultwarden. It will read the `.env` from the cwd, not from the data folder. Or you need to configure the `ENV_FILE` variable to point it to the correct location.
